### ISSUE TITLE
Display of CallbackError

### ISSUE THREAD

**Issue by TimUntersberger** [khvzak/mlua#71]
https://github.com/khvzak/mlua/blob/7e7a44f4cd16600f2ce0d72204859d0ad35995c8/src/error.rs#L249

Is there a reason why the `std::fmt::Display` implementation for `Error::CallbackError` does not include the cause?

Ideally I would prefer having something like the following:

```lua
            Error::CallbackError { ref traceback, cause } => {
                write!(fmt, "callback error: {}\n{}", cause, traceback)
            }
```

or

```lua
            Error::CallbackError { ref traceback, cause } => {
                write!(fmt, "callback error: \ncause:{}\n{}", cause, traceback)
            }
```

**Comment by khvzak** [comment]
This is historically inherited from rlua (see https://github.com/amethyst/rlua/pull/167)

Personally I don't have strong opinion on this.

**Comment by chipsenkbeil** [comment]
@khvzak I just encountered a callback error as well and the lack of clarity makes it a little more difficult to debug. Would be handy to have more information displayed.

**Comment by khvzak** [comment]
Agree, will find a solution.

Just prining `cause` is be enough probably, because the following code:
```rust
        let f = lua.create_function(|_, ()| Err::<(), _>("bla".to_lua_err()))?;
        let g = lua.create_function(|lua, ()| {
            lua.globals().get::<_, LuaFunction>("f")?.call::<_, ()>(())
        })?;
        lua.globals().set("f", f)?;
        lua.globals().set("g", g)?;
        let err = lua.load("g()").exec().err().unwrap();
        println!("{}", err);
```

would produce the following:
```
callback error: callback error: bla
stack traceback:
	[C]: in function 'f'
	[C]: in function 'g'
	[string "src/main.rs:27:14"]:1: in main chunk
stack traceback:
	[C]: in function 'g'
	[string "src/main.rs:27:14"]:1: in main chunk
```
which is confusing.

**Comment by khvzak** [comment]
I pushed fix to the dev branch (will be included in v0.7) to nicely format errors and incude source to the Display implementation.
